compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/inlineclasses/InlineClassAbi.kt

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.backend.jvm.lower.inlineclasses
+
+import org.jetbrains.kotlin.backend.jvm.ir.erasedUpperBound
+import org.jetbrains.kotlin.ir.declarations.*
+import org.jetbrains.kotlin.ir.types.IrType
+import org.jetbrains.kotlin.ir.types.classOrNull
+import org.jetbrains.kotlin.ir.types.isPrimitiveType
+import org.jetbrains.kotlin.ir.types.makeNullable
+import org.jetbrains.kotlin.ir.util.*
+import org.jetbrains.kotlin.load.java.JvmAbi
+import org.jetbrains.kotlin.name.Name
+import java.security.MessageDigest
+import java.util.*
+
+/**
+ * Replace inline classes by their underlying types.
+ */
+fun IrType.unbox() = InlineClassAbi.unboxType(this) ?: this
+
+/**
+ * Check if a function can be replaced.
+ */
+val IrFunction.isBoxed: Boolean
```

Comment:
Maybe consider naming this `hasInlineClassParameters` or something more explicit, or make it `internal`.

Also I didn't understand the kdoc

Reply:
Done.